packages/back-end/src/util/sql.ts

```diff
@@ -209,19 +209,80 @@ export function replaceCountStar(aggregation: string, col: string) {
   return aggregation.replace(/count\(\s*\*\s*\)/gi, `COUNT(${col})`);
 }
 
+function isJSON(str: string) {
+  // Only match objects
+  if (!str?.startsWith("{")) return false;
+
+  try {
+    JSON.parse(str);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function getJSONFields(
+  testValues: unknown[]
+): Record<string, FactTableColumnType> {
+  const fields: Record<string, FactTableColumnType> = {};
+
+  testValues.forEach((str) => {
+    if (typeof str !== "string") return;
+    try {
+      const obj = JSON.parse(str);
+      Object.keys(obj).forEach((key) => {
+        if (fields[key]) return;
+        if (obj[key] === null || obj[key] === undefined) return;
+        if (Object.keys(fields).length > 50) return;
```

Comment:
Any particular reason for this limit of 50 fields? Just too unwieldy to use effectively? Likely to be garbage?

Reply:
Since these fields are stored in the fact table itself and we pass that around everywhere, trying to add some limits to the size of that object.